app/ui/src/app/integration/integration.model.ts

```diff
@@ -0,0 +1,125 @@
+  configuredProperties: {};
+  stepKind: string;
+  id: string;
+}
+export type Steps = Array<Step>;
+
+export interface Integration extends BaseEntity {
+  configuration: string;
+  description: string;
+  deployedRevisionId: number;
+  revisions: Array<IntegrationRevision>;
+  statusMessage: string;
+  token: string;
+  steps: Array<Step>;
+  draftRevision: IntegrationRevision;
+  gitRepo: string;
+  users: Array<User>;
+  connections: Array<Connection>;
+  userId: string;
+  desiredStatus: 'Draft' | 'Pending' | 'Activated' | 'Deactivated' | 'Deleted';
+  currentStatus: 'Draft' | 'Pending' | 'Activated' | 'Deactivated' | 'Deleted';
+  stepsDone: Array<string>;
+  lastUpdated: string;
+  createdDate: string;
+  timesUsed: number;
+  integrationTemplateId: string;
+  id: string;
+  tags: Array<string>;
+  name: string;
+}
+export type Integrations = Array<Integration>;
+
+export interface IntegrationRevision extends BaseEntity {
+  currentMessage: string;
+  parentVersion: number;
+  targetState:
+    | 'Draft'
+    | 'Active'
+    | 'Inactive'
+    | 'Undeployed'
+    | 'Error'
+    | 'Pending';
+  targetMessage: string;
+  version: number;
+  spec: IntegrationRevisionSpec;
+  currentState:
+    | 'Draft'
+    | 'Active'
+    | 'Inactive'
+    | 'Undeployed'
+    | 'Error'
+    | 'Pending';
+}
+export type IntegrationRevisions = Array<IntegrationRevision>;
+
+export interface IntegrationRevisionSpec extends BaseEntity {
+  configuration: string;
+  steps: Array<Step>;
+  connections: Array<Connection>;
+}
+export type IntegrationRevisionSpecs = Array<IntegrationRevisionSpec>;
+
+export function createStep() {
+    return <Step>{
+      action: undefined,
+      connection: undefined,
+      name: undefined,
+      configuredProperties: undefined,
+      stepKind: undefined,
+      id: undefined
+    };
+  }
+
+export function createIntegration() {
+  return <Integration>{
+    configuration: undefined,
+    description: undefined,
+    deployedRevisionId: undefined,
+    revisions: undefined,
+    statusMessage: undefined,
+    token: undefined,
+    steps: undefined,
+    draftRevision: undefined,
+    gitRepo: undefined,
+    users: undefined,
+    connections: undefined,
+    userId: undefined,
+    desiredStatus: undefined,
+    currentStatus: undefined,
+    stepsDone: undefined,
+    lastUpdated: undefined,
+    createdDate: undefined,
+    timesUsed: undefined,
+    integrationTemplateId: undefined,
+    id: undefined,
+    tags: undefined,
+    name: undefined
+  };
+}
+
+export function createIntegrationRevision() {
+  return <IntegrationRevision>{
+    currentMessage: undefined,
+    parentVersion: undefined,
+    targetState: undefined,
+    targetMessage: undefined,
+    version: undefined,
+    spec: undefined,
+    currentState: undefined
+  };
+}
+
+export function createIntegrationRevisionSpec() {
+  return <IntegrationRevisionSpec>{
+    configuration: undefined,
+    steps: undefined,
+    connections: undefined
+  };
+}
```

Comment:
This is the only section of the entire PR that I believe needs to be addressed in a different way: Having all the functions returning object instances where all properties are set to undefined are simply unnecessary and ad a lot of boilerplate to the codebase. If we eventually move these functions elsewhere or either any of the types returned or its ancestors change its definition, the function will make the compiler complain. 

**Solution**: If a type (any type) is meant to support nullable/non-nullable properties mapped as `undefined`, then the original type property should be defined as optional by adding a `?` to the property definition. 

Example - Refactor the `Step` interface like this (please notice that its ancestor has been removed):

```typescript
export interface Step {
   action?: Action;
   connection?: Connection;
   name?: string;
   configuredProperties?: any;
   stepKind?: string;
   id?: string;
 }
``` 

Then you can refactor the `createStep` function like this, in a type safe mode (unless you further update `Step` to include a mandatory, non-nullable typed property):

```typescript
export const createStep = () => ({} as Step);
```

You can even go one step further and do this on a more generic way:

```typescript
export function integrationTypeFactory<T>(): T {
   return {} as T;
}
```

This would allow to replace all the 3 functions above at once.